### PR TITLE
Add kotlin docs

### PR DESCRIPTION
**Pull request by jan-tennert**
## What kind of change does this PR introduce?

This PR adds documentation for `supabase-kt`.
I also changed some outdated links for the old kotlin repositories (postgrest-kt and gotrue-kt)

## What is the current behavior?

No kotlin docs

## What is the new behavior?

New kotlin docs

## Additional context

I'm still not completely done with the actual documentation, but it's nearly complete.

### CODE REVIEW

**Comment by saltcod** [comment]
Amazing contribution @jan-tennert !   Will have a look!

File: i18n/README.ar.md [hunk 1]
```diff
@@ -134,11 +134,12 @@ Supabase عبارة عن مجموعة من الأدوات مفتوحة المص
   </tr>
   <tr>
     <td>Kotlin</td>
-    <td>-</td>
-    <td><a href="https://github.com/supabase-community/postgrest-kt" target="_blank" rel="noopener noreferrer">postgrest-kt</a></td>
-    <td><a href="https://github.com/supabase-community/gotrue-kt" target="_blank" rel="noopener noreferrer">gotrue-kt</a></td>
-    <td>-</td>
-    <td>-</td>
+    <td><a href="https://github.com/supabase-community/supabase-kt" target="_blank" rel="noopener noreferrer">supabase-kt</a></td>
+    <td><a href="https://github.com/supabase-community/supabase-kt/tree/master/Postgrest" target="_blank" rel="noopener noreferrer">realtime-kt</a></td>
```
Comment (kevcodez) [inline i18n/README.ar.md:138]:
```suggestion
    <td><a href="https://github.com/supabase-community/supabase-kt/tree/master/Postgrest" target="_blank" rel="noopener noreferrer">postgrest-kt</a></td>
```

File: i18n/README.ar.md [hunk 2]
```diff
@@ -134,11 +134,12 @@ Supabase عبارة عن مجموعة من الأدوات مفتوحة المص
...
+    <td><a href="https://github.com/supabase-community/supabase-kt/tree/master/GoTrue" target="_blank" rel="noopener noreferrer">realtime-kt</a></td>
```
Comment (kevcodez) [inline i18n/README.ar.md:139]:
```suggestion
    <td><a href="https://github.com/supabase-community/supabase-kt/tree/master/GoTrue" target="_blank" rel="noopener noreferrer">gotrue-kt</a></td>
```

File: i18n/README.ar.md [hunk 3]
```diff
@@ -134,11 +134,12 @@ Supabase عبارة عن مجموعة من الأدوات مفتوحة المص
...
+    <td><a href="https://github.com/supabase-community/supabase-kt/tree/master/Realtime" target="_blank" rel="noopener noreferrer">realtime-kt</a></td>
+    <td><a href="https://github.com/supabase-community/supabase-kt/tree/master/Storage" target="_blank" rel="noopener noreferrer">storage-kt</a></td>
+    <td><a href="https://github.com/supabase-community/supabase-kt/tree/master/Functions" target="_blank" rel="noopener noreferrer">functions-kt</a></td>
   </tr>
```
Comment (kevcodez) [inline i18n/README.ar.md:143]: Please also adjust the links in the other README.*.md files

File: i18n/README.fr.md [hunk 4]
```diff
@@ -138,8 +138,8 @@ Notre bibliothèque client est modulaire. Chaque sous-bibliothèque est une impl
   <tr>
     <td>Kotlin</td>
     <td>-</td>
```
Comment (kevcodez) [inline i18n/README.fr.md:140]: Could you fill out the remaining items and make sure the name fits the link?

File: i18n/README.fi.md [hunk 5]
```diff
@@ -147,12 +147,12 @@ Lähestymistapamme asiakaskirjastoissa on modulaarinen. Jokainen alikirjasto on
   </tr>
   <tr>
     <td>Kotlin</td>
-    <td>-</td>
-    <td><a href="https://github.com/supabase-community/postgrest-kt" target="_blank" rel="noopener noreferrer">postgrest-kt</a></td>
-    <td><a href="https://github.com/supabase-community/gotrue-kt" target="_blank" rel="noopener noreferrer">gotrue-kt</a></td>
-    <td>-</td>
-    <td>-</td>
-    <td>-</td>
+    <td><a href="https://github.com/supabase-community/supabase-kt" target="_blank" rel="noopener noreferrer">supabase-kt</a></td>
```
Comment (kevcodez) [inline i18n/README.fi.md:150]: Could you make sure the name fits the link?

File: i18n/README.hi.md [hunk 6]
```diff
@@ -141,8 +141,8 @@
   <tr>
     <td>कोटलिन</td>
     <td>-</td>
```
Comment (kevcodez) [inline i18n/README.hi.md:143]: Could you fill out the remaining items and make sure the name fits the link?

File: i18n/README.jp.md [hunk 7]
```diff
@@ -147,12 +147,12 @@ Supabase クライアントライブラリはモジュール化されていま
   </tr>
   <tr>
     <td>Kotlin</td>
-    <td>-</td>
-    <td><a href="https://github.com/supabase-community/postgrest-kt" target="_blank" rel="noopener noreferrer">postgrest-kt</a></td>
-    <td><a href="https://github.com/supabase-community/gotrue-kt" target="_blank" rel="noopener noreferrer">gotrue-kt</a></td>
-    <td>-</td>
-    <td>-</td>
-    <td>-</td>
+    <td><a href="https://github.com/supabase-community/supabase-kt" target="_blank" rel="noopener noreferrer">supabase-kt</a></td>
```
Comment (kevcodez) [inline i18n/README.jp.md:150]: Could you make sure the name fits the link?

File: i18n/README.ko.md [hunk 8]
```diff
@@ -145,12 +145,12 @@ Supabase 클라이언트 라이브러리는 모듈식 입니다. 각 하위 라
   </tr>
   <tr>
     <td>Kotlin</td>
-    <td>-</td>
-    <td><a href="https://github.com/supabase-community/postgrest-kt" target="_blank" rel="noopener noreferrer">postgrest-kt</a></td>
-    <td><a href="https://github.com/supabase-community/gotrue-kt" target="_blank" rel="noopener noreferrer">gotrue-kt</a></td>
-    <td>-</td>
-    <td>-</td>
-    <td>-</td>
+    <td><a href="https://github.com/supabase-community/supabase-kt" target="_blank" rel="noopener noreferrer">supabase-kt</a></td>
```
Comment (kevcodez) [inline i18n/README.ko.md:148]: Could you make sure the name fits the link?

File: i18n/README.pt-br.md [hunk 9]
```diff
@@ -148,8 +148,8 @@ Nossa biblioteca de cliente é modular. Cada sub-biblioteca é uma implementaç
   <tr>
     <td>Kotlin</td>
     <td><a href="https://github.com/supabase-community/supabase-kt" target="_blank" rel="noopener noreferrer">supabase-kt</a></td>
```
Comment (kevcodez) [inline i18n/README.pt-br.md:150]: Could you make sure the name fits the link?

File: i18n/README.sr.md [hunk 10]
```diff
@@ -147,12 +147,12 @@ Naš pristup klijentskim bibliotekama je modularan. Svaka pod-biblioteka je samo
   </tr>
   <tr>
     <td>Kotlin</td>
-    <td>-</td>
-    <td><a href="https://github.com/supabase-community/postgrest-kt" target="_blank" rel="noopener noreferrer">postgrest-kt</a></td>
-    <td><a href="https://github.com/supabase-community/gotrue-kt" target="_blank" rel="noopener noreferrer">gotrue-kt</a></td>
-    <td>-</td>
-    <td>-</td>
-    <td>-</td>
+    <td><a href="https://github.com/supabase-community/supabase-kt" target="_blank" rel="noopener noreferrer">supabase-kt</a></td>
```
Comment (kevcodez) [inline i18n/README.sr.md:150]: Could you make sure the name fits the link?

File: i18n/README.tr.md [hunk 11]
```diff
@@ -135,8 +135,8 @@ Müşteri kitaplığımız modülerdir. Her bir alt kütüphane, tek bir harici
   <tr>
     <td>Kotlin</td>
     <td>-</td>
-    <td><a href="https://github.com/supabase-community/postgrest-kt" target="_blank" rel="noopener noreferrer">postgrest-kt</a></td>
-    <td><a href="https://github.com/supabase-community/gotrue-kt" target="_blank" rel="noopener noreferrer">gotrue-kt</a></td>
+    <td><a href="https://github.com/supabase-community/supabase-kt/tree/master/Postgrest" target="_blank" rel="noopener noreferrer">realtime-kt</a></td>
```
Comment (kevcodez) [inline i18n/README.tr.md:138]: Could you fill out the remaining items and make sure the name fits the link?

File: package-lock.json [hunk 12]
```diff
@@ -70213,4 +70213,4 @@
       "integrity": "sha512-Bq0B+ixT/DMyG8kgX2xWcI5jUvCwqrMxSFam7m0lAf78nf04hv6lNCsyLYdyYTrCVMqNDY/206K7eExYCeSyUQ=="
     }
   }
-}
+}
```
Comment (kevcodez) [inline package-lock.json:70216]: Could you please revert this change?

File: packages/common/constants/client-libraries.ts [hunk 13]
```diff
@@ -133,11 +133,23 @@ export const CLIENT_LIBRARIES = [
     libraries: [
       {
         name: 'postgrest-kt',
-        url: 'https://github.com/supabase-community/postgrest-kt',
+        url: 'https://github.com/supabase-community/supabase-kt/Postgrest',
```
Comment (kevcodez) [inline packages/common/constants/client-libraries.ts:136]: These urls all lead to 404

File: i18n/README.ta.md [hunk 14]
```diff
@@ -142,11 +142,12 @@
   </tr>
   <tr>
     <td>கோட்லின்</td>
-    <td>-</td>
-    <td><a href="https://github.com/supabase-community/postgrest-kt" target="_blank" rel="noopener noreferrer">போஸ்ட்ஜீராஸ்ட்-கேடி</a></td>
-    <td><a href="https://github.com/supabase-community/gotrue-kt" target="_blank" rel="noopener noreferrer">கூட்ரூ-கேடி</a></td>
-    <td>-</td>
-    <td>-</td>
+    <td><a href="https://github.com/supabase-community/supabase-kt" target="_blank" rel="noopener noreferrer">supabase-kt</a></td>
```
Comment (kevcodez) [inline i18n/README.ta.md:145]: Could you make sure the name fits the link?

File: i18n/README.vi-vn.md [hunk 15]
```diff
@@ -147,12 +147,12 @@ Phương pháp tiếp cận của chúng tôi cho các thư viện hỗ trợ l
   </tr>
   <tr>
     <td>Kotlin</td>
-    <td>-</td>
-    <td><a href="https://github.com/supabase-community/postgrest-kt" target="_blank" rel="noopener noreferrer">postgrest-kt</a></td>
-    <td><a href="https://github.com/supabase-community/gotrue-kt" target="_blank" rel="noopener noreferrer">gotrue-kt</a></td>
-    <td>-</td>
-    <td>-</td>
-    <td>-</td>
+    <td><a href="https://github.com/supabase-community/supabase-kt" target="_blank" rel="noopener noreferrer">supabase-kt</a></td>
```
Comment (kevcodez) [inline i18n/README.vi-vn.md:150]: Could you make sure the name fits the link?

File: i18n/README.zh-cn.md [hunk 16]
```diff
@@ -141,11 +141,12 @@ Supabase 是一个[托管平台](https://app.supabase.com)。你可以注册并
   </tr>
   <tr>
     <td>Kotlin</td>
-    <td>-</td>
-    <td><a href="https://github.com/supabase-community/postgrest-kt" target="_blank" rel="noopener noreferrer">postgrest-kt</a></td>
-    <td><a href="https://github.com/supabase-community/gotrue-kt" target="_blank" rel="noopener noreferrer">gotrue-kt</a></td>
-    <td>-</td>
-    <td>-</td>
+    <td><a href="https://github.com/supabase-community/supabase-kt" target="_blank" rel="noopener noreferrer">supabase-kt</a></td>
```
Comment (kevcodez) [inline i18n/README.zh-cn.md:144]: Could you make sure the name fits the link?

File: i18n/README.zh-tw.md [hunk 17]
```diff
@@ -146,8 +146,8 @@ Supabase 是[託管平台](https://app.supabase.com)。只要註冊，不必安
   <tr>
     <td>Kotlin</td>
     <td><a href="https://github.com/supabase-community/supabase-kt" target="_blank" rel="noopener noreferrer">supabase-kt</a></td>
-    <td><a href="https://github.com/supabase-community/postgrest-kt" target="_blank" rel="noopener noreferrer">postgrest-kt</a></td>
-    <td><a href="https://github.com/supabase-community/gotrue-kt" target="_blank" rel="noopener noreferrer">gotrue-kt</a></td>
+    <td><a href="https://github.com/supabase-community/supabase-kt/tree/master/Postgrest" target="_blank" rel="noopener noreferrer">realtime-kt</a></td>
```
Comment (kevcodez) [inline i18n/README.zh-tw.md:149]: Could you make sure the name fits the link?

File: i18n/README.es.md [hunk 18]
```diff
@@ -147,8 +147,7 @@ Nuestra librería de cliente es modular. Cada sub-librería es una implementaci
   <tr>
     <td>Kotlin</td>
     <td><a href="https://github.com/supabase-community/supabase-kt" target="_blank" rel="noopener noreferrer">supabase-kt</a></td>
-    <td><a href="https://github.com/supabase-community/postgrest-kt" target="_blank" rel="noopener noreferrer">postgrest-kt</a></td>
-    <td><a href="https://github.com/supabase-community/gotrue-kt" target="_blank" rel="noopener noreferrer">gotrue-kt</a></td>
+    <td><a href="https://github.com/supabase-community/supabase-kt/tree/master/Realtime" target="_blank" rel="noopener noreferrer">realtime-kt</a></td>
```
Comment (kevcodez) [inline i18n/README.es.md:150]: I think this got messed up a little. Realtime is in there twice now, postgrest and gotrue are missing

File: i18n/README.bg.md [hunk 19]
```diff
@@ -141,11 +141,12 @@ Supabase е [хоствана платформа](https://app.supabase.com). М
   </tr>
   <tr>
     <td>Kotlin</td>
-    <td>-</td>
-    <td><a href="https://github.com/supabase-community/postgrest-kt" target="_blank" rel="noopener noreferrer">postgrest-kt</a></td>
-    <td><a href="https://github.com/supabase-community/gotrue-kt" target="_blank" rel="noopener noreferrer">gotrue-kt</a></td>
-    <td>-</td>
-    <td>-</td>
+    <td><a href="https://github.com/supabase-community/supabase-kt" target="_blank" rel="noopener noreferrer">supabase-kt</a></td>
```
Comment (kevcodez) [inline i18n/README.bg.md:144]: Could you make sure the name fits the link?

File: apps/docs/docs/ref/kotlin/introduction.mdx [hunk 20]
```diff
@@ -0,0 +1,33 @@
...
+    <h1 className="text-3xl text-scale-1200 m-0">Kotlin Client Library</h1>
+    <h2 className="text-base font-mono text-scale-1100">@supabase-community/supabase-kt</h2>
+  </div>
+</div>
+
+<div className="max-w-xl">
+This reference documents every object and method available in Supabase's Kotlin Multiplatform library, [supabase-kt](https://github.com/supabase-community/supabase-kt). You can use supabase-kt to interact with your Postgres database, listen to database changes, invoke Deno Edge Functions, build login and user management functionality, and manage large files.
+
+Supported Kotlin targets:
+    - JVM (Desktop)
```
Comment (kevcodez) [inline apps/docs/docs/ref/kotlin/introduction.mdx:19]: Isn't JVM target universal (so also server environment?) If yes, I'd remove the suffix

File: apps/docs/docs/ref/kotlin/installing.mdx [hunk 21]
```diff
@@ -0,0 +1,267 @@
+---
+id: installing
+title: 'Installing & Initialization'
+slug: installing
+custom_edit_url: https://github.com/supabase/supabase/edit/master/web/spec/supabase.yml
```
Comment (kevcodez) [inline apps/docs/docs/ref/kotlin/installing.mdx:5]: @gregnr Should we leave this one out? Probably makes little sense to link to the `supabase.yml` other `installing.mdx` do this too though

File: apps/docs/docs/ref/kotlin/installing.mdx [hunk 22]
```diff
@@ -0,0 +1,267 @@
...
+                }
+                ```
+
+            </TabPanel>
+        </Tabs>
+
+    </RefSubLayout.Examples>
+</RefSubLayout.EducationRow>
+
+### Add KotlinX Serialization (required for postgrest and realtime serialization)
```
Comment (kevcodez) [inline apps/docs/docs/ref/kotlin/installing.mdx:184]:
```suggestion
### Add KotlinX Serialization (required for PostgREST and Realtime serialization)

```

Comment (kevcodez) [inline apps/docs/docs/ref/kotlin/installing.mdx:184]:
```suggestion
### Add KotlinX Serialization (required for PostgREST and Realtime serialization)

```

File: spec/supabase_kt_v0.yml [hunk 23]
```diff
@@ -0,0 +1,3502 @@
...
+          
+          `httpPort`: The port the web server is running on, when logging in with OAuth. Default: `0` (random port).
+          
+          `timeout`: The timeout for the web server, when logging in with OAuth. Default: `1.minutes`.
+          
+          `htmlTitle`: The title of the redirect page, when logging in with OAuth. Default: `"Supabase Auth"`.
+          
+          `htmlText`: The text of the redirect page, when logging in with OAuth. Default: `"Logged in. You may continue in your app."`.
+          
+          `htmlIconUrl`: The icon of the redirect page, when logging in with OAuth. Default: `"https://supabase.com/brand-assets/supabase-logo-icon.png"`.
```
Comment (kevcodez) [inline spec/supabase_kt_v0.yml:95]: This leads to a 404 - @gregnr @saltcod do we have a static URL for our logo?

File: spec/supabase_kt_v0.yml [hunk 24]
```diff
@@ -0,0 +1,3502 @@
...
+                  propertyConversionMethod = PropertyConversionMethod.SERIAL_NAME // default: PropertyConversionMethod.CAMEL_CASE_TO_SNAKE_CASE
+              }
+          }
+          ```
+        description: |
+          `propertyConversionMethod` - The method to use to convert the property names to the column names when applying filters & using the update method. Default: `PropertyConversionMethod.CAMEL_CASE_TO_SNAKE_CASE`
+          
+          `defaultSchema` - The default schema to use for database requests. Default: `public`
+          
+          `customUrl` - Custom url for the Postgrest Api. Can be safely ignored when using supabase. Default: `null`
```
Comment (kevcodez) [inline spec/supabase_kt_v0.yml:115]:
```suggestion
          `customUrl` - Custom url for the PostgREST Api. Can be safely ignored when using Supabase. Default: `null`
```

File: spec/supabase_kt_v0.yml [hunk 25]
```diff
@@ -0,0 +1,3502 @@
...
+      - id: configure-realtime
+        name: Configure Realtime module
+        code: |
+          ```kotlin
+          val client = createSupabaseClient(
+              supabaseUrl = "https://xyzcompany.supabase.co", 
+              supabaseKey = "public-anon-key"
+          ) {
+              install(Realtime) {
+                  reconnectDelay = 5.seconds //Default: 7 seconds
```
Comment (kevcodez) [inline spec/supabase_kt_v0.yml:154]:
```suggestion
                  reconnectDelay = 5.seconds // Default: 7 seconds
```

File: spec/supabase_kt_v0.yml [hunk 26]
```diff
@@ -0,0 +1,3502 @@
...
+          
+          `resumable.cache` - Interface for storing resumable upload urls. Default: `ResumableCache.Disk`
+          
+          `resumable.defaultChunkSize` - The default chunk size for resumable uploads. Supabase currently only supports a chunk size of 6MB, so be careful when changing this value. Default: `6MB`
+          
+          `resumable.retryTimeout` - the timeout for retrying resumable uploads when uploading a chunk fails. Default: `5.seconds`
+          
+          `resumable.onlyUpdateStateAfterChunk` - whether the upload state should only be updated after a chunk was uploaded successfully or also when the chunk is currently being uploaded. Default: `false`
+          
+          `customUrl` - Custom url for the Storage Api. Can be safely ignored when using supabase. Default: `null`
```
Comment (kevcodez) [inline spec/supabase_kt_v0.yml:142]:
```suggestion
          `customUrl` - Custom url for the Storage Api. Can be safely ignored when using Supabase. Default: `null`
```

File: spec/supabase_kt_v0.yml [hunk 27]
```diff
@@ -0,0 +1,3502 @@
...
+              supabaseUrl = "https://xyzcompany.supabase.co", 
+              supabaseKey = "public-anon-key"
+          ) {
+              install(Realtime) {
+                  reconnectDelay = 5.seconds //Default: 7 seconds
+              }
+          }
+          ```
+        description: |
+          `reconnectDelay` - The delay between reconnect attempts. Default: `7 seconds`
```
Comment (kevcodez) [inline spec/supabase_kt_v0.yml:159]: Format of the time seems a bit inconsistent, sometimes it is "x.seconds", sometimes it is "x seconds"

File: i18n/README.es.md (hunk 18 above)
Comment (jan-tennert) [reply i18n/README.es.md]: Yep, that happens when trying to update non-matching READMEs in bulk. Should be fixed now

File: spec/supabase_kt_v0.yml [hunk 28]
```diff
@@ -0,0 +1,3502 @@
...
+          
+          `heartbeatInterval` - The interval between heartbeat messages. Default: `15 seconds`
+          
+          `disconnectOnSessionLoss` - Whether to disconnect from the websocket when the session is lost. Default: `true`
+          
+          `secure` - Whether to use wss or ws. Defaults to [SupabaseClient.useHTTPS] when null
+          
+          `websocketConfig` - Custom ktor websocket config
```
Comment (kevcodez) [inline spec/supabase_kt_v0.yml:167]:
```suggestion
          `websocketConfig` - Custom Ktor websocket config
```

File: spec/supabase_kt_v0.yml [hunk 29]
```diff
@@ -0,0 +1,3502 @@
...
+          
+          `customUrl` - Custom url for the Realtime Websocket. Can be safely ignored when using supabase. Default: `null`
```
Comment (kevcodez) [inline spec/supabase_kt_v0.yml:169]:
```suggestion
          `customUrl` - Custom url for the Realtime websocket. Can be safely ignored when using Supabase. Default: `null`
```

File: spec/supabase_kt_v0.yml [hunk 30]
```diff
@@ -0,0 +1,3502 @@
...
+              supabaseUrl = "https://xyzcompany.supabase.co", 
+              supabaseKey = "public-anon-key"
+          ) {
+              install(Functions) {
+                  //no custom settings
+              }
+          }
+          ```
+        description: |
+          `customUrl` - Custom url for the Functions Api. Can be safely ignored when using supabase. Default: `null`
```
Comment (kevcodez) [inline spec/supabase_kt_v0.yml:186]:
```suggestion
          `customUrl` - Custom url for the Functions API. Can be safely ignored when using Supabase. Default: `null`
```

Comment (kevcodez) [inline spec/supabase_kt_v0.yml:186]:
```suggestion
          `customUrl` - Custom url for the Functions API. Can be safely ignored when using Supabase. Default: `null`
```

File: spec/supabase_kt_v0.yml [hunk 31]
```diff
@@ -0,0 +1,3502 @@
...
+                  apolloConfiguration {
+                    //custom configuration
+                  }
+              }
+          }
+          ```
+        description: |
+          `apolloConfiguration` - Custom configuration for the ApolloClient
+          
+          `customUrl` - Custom url for the GraphQL Api. Can be safely ignored when using supabase. Default: `null`
```
Comment (kevcodez) [inline spec/supabase_kt_v0.yml:207]:
```suggestion
          `customUrl` - Custom url for the GraphQL Api. Can be safely ignored when using Supabase. Default: `null`
```

File: spec/supabase_kt_v0.yml [hunk 32]
```diff
@@ -0,0 +1,3502 @@
...
+          
+          `jwtToken` - Plugin specific JWT Token. Can be ignored when using the GoTrue plugin. Default: `null`
+
+  - id: select
+    title: 'Fetch data: select()'
+    notes: |
+      Perform a SELECT query on the table or view.
+      - When calling a `decode` method, you have to provide a [serializable class](/docs/reference/kotlin/installing#add-kotlinx-serialization-required-for-postgrest-and-realtime-serialization) as the type parameter.
+      - You can provide a `Columns` object to select specific columns.
+      - You can provide a [filter]() block to filter the results
```
Comment (kevcodez) [inline spec/supabase_kt_v0.yml:217]: Is this supposed to link anywhere?

File: spec/supabase_kt_v0.yml [hunk 33]
```diff
@@ -0,0 +1,3502 @@
...
+        isSpotlight: true
+        code: |
+          ```kotlin
+          val toUpsert = Message(id = 3, message = "foo", username = "supabot")
+          client.postgrest["messages"].insert(toUpsert, upsert = true)
+          ```
+      - id: upserting-into-tables-with-constraints
+        name: Upserting into tables with constraints
+        description: |
+          Running the following will cause supabase to upsert data into the `users` table.
```
Comment (kevcodez) [inline spec/supabase_kt_v0.yml:449]:
```suggestion
          Running the following will cause Supabase to upsert data into the `users` table.
```

File: spec/supabase_kt_v0.yml [hunk 34]
```diff
@@ -0,0 +1,3502 @@
...
+    title: 'signUp()'
+    $ref: '@supabase/gotrue-js.GoTrueClient.signUp'
+    notes: |
+      - By default, the user needs to verify their email address before logging in. To turn this off, disable **Confirm email** in [your project](https://app.supabase.com/project/_/auth/providers).
+      - **Confirm email** determines if users need to confirm their email address after signing up.
+        - If **Confirm email** is enabled, the return value is null and you will be logged in instead.
+        - If **Confirm email** is disabled, the return value is the user and you won't be logged in automatically.
+      - When the user confirms their email address, they are redirected to the [`SITE_URL`](https://supabase.com/docs/reference/auth/config#site_url) by default. You can modify your `SITE_URL` or add additional redirect URLs in [your project](https://app.supabase.com/project/_/auth/url-configuration).
+      - supabase-kt has default platform implementations: [deep links](https://developer.android.com/training/app-links/deep-linking) on Android, [url schemes](https://developer.apple.com/documentation/xcode/defining-a-custom-url-scheme-for-your-app) on IOS and the website on JS
```
Comment (kevcodez) [inline spec/supabase_kt_v0.yml:2045]: might make sense to add a link to github on supabase-kt

File: spec/supabase_kt_v0.yml [hunk 35]
```diff
@@ -0,0 +1,3502 @@
...
+    notes: |
+      - Before you can call this method you need to [establish a connection](/docs/guides/auth/sso/auth-sso-saml#managing-saml-20-connections) to an identity provider. Use the [CLI commands](/docs/reference/cli/supabase-sso) to do this.
+      - If you've associated an email domain to the identity provider, you can use the `SSO.withDomain()` function in the `loginWith` method to start a sign-in flow.
+      - In case you need to use a different way to start the authentication flow with an identity provider, you can use the `SSO.withProviderId` function. For example:
+          - Mapping specific user email addresses with an identity provider.
+          - Using different hints to identity the identity provider to be used by the user, like a company-specific page, IP address or other tracking information.
+      - supabase-kt has default platform implementations: HTTP Callback Server on Desktop, [deep links](https://developer.android.com/training/app-links/deep-linking) on Android, [url schemes](https://developer.apple.com/documentation/xcode/defining-a-custom-url-scheme-for-your-app) on IOS and the website on JS
```
Comment (kevcodez) [inline spec/supabase_kt_v0.yml:2235]: might make sense to add a link to github on supabase-kt

File: spec/supabase_kt_v0.yml [hunk 36]
```diff
@@ -0,0 +1,3502 @@
...
+          
+          `flowType` - Which authentication flow to use. Currently available: FlowType.PKCE and FlowType.IMPLICIT. Default: `FlowType.IMPLICIT`
+          
+          `codeVerifierCache` - Interface for saving and loading codes for the PKCE authentication flow. Default: `SettingsCodeVerifierCache`
+          
+          `customUrl` - Custom url for the GoTrue Api. Can be safely ignored when using supabase. Default: `null`
+          
+          `jwtToken` - Plugin specific JWT Token. Can be ignored when using the GoTrue plugin. Default: `null`
+          
+          `retryDelay` - Duration after which the GoTrue plugin should retry a failed session refresh. Default: `10.seconds`
```
[the remark this answers is not in the document]
Comment (jan-tennert) [reply spec/supabase_kt_v0.yml:71]: Yes, it's from Kotlin's standard library. Wasn't sure to put this or `10 seconds`

File: spec/supabase_kt_v0.yml (hunk 35 above)
[the remark this answers is not in the document]
Comment (jan-tennert) [reply spec/supabase_kt_v0.yml]: I mean it's both, I replaced it with JVM for now to be consistent with the targets

**Comment by jan-tennert** [comment]
Also sometimes a few letters get cut of, I assume this is because of the codeblocks (?). In this case `js`:
image
Or here `ts`:
image
Doesn't happen in strings though.

File: apps/docs/docs/ref/kotlin/installing.mdx [hunk 37]
```diff
@@ -0,0 +1,267 @@
...
+custom_edit_url: https://github.com/supabase/supabase/edit/master/web/spec/supabase.yml
+---
+
+
+### Add one or more modules to your project
+
+<RefSubLayout.EducationRow>
+    <RefSubLayout.Examples>
+        <div style={{ display: 'flex', flexDirection: 'row', alignItems: 'center' }}>
+            <p style={{ marginRight: '8px' }}>Newest version:</p>
```
Comment (kevcodez) [inline apps/docs/docs/ref/kotlin/installing.mdx:14]: Nit: You could skip the "Newest version", I believe the two images should be sufficient

**Comment by gregnr** [comment]
> Also sometimes a few letters get cut of, I assume this is because of the codeblocks (?)

Looks like it has to do with:
https://github.com/supabase/supabase/blob/b550f975a2b141ede38ee9285339bdcf5c45bf37/apps/docs/components/reference/RefFunctionSection.tsx#L155-L159

Definitely a bug, lets address in another PR though.

**Comment by kevcodez** [comment]
There are a few items that miss a description

- Unenroll a factor
- Generate an email link
- Update a user
- List all factors for a user
- Delete a factor for a user

----

The Kotlin logo seems to be slightly misaligned, maybe @gregnr can have a look

Screenshot 2023-05-10 at 20 01 28

----

Looking at the JS docs, they use params like below

Screenshot 2023-05-10 at 20 07 15

whereas the Kotlin docs seem to use the "Notes" field to describe parameters/options

**Comment by jan-tennert** [comment]
> There are a few items that miss a description
> 
> * Unenroll a factor
> * Generate an email link
> * Update a user
> * List all factors for a user
> * Delete a factor for a user
> 
> The Kotlin logo seems to be slightly misaligned, maybe @gregnr can have a look
> 
> Screenshot 2023-05-10 at 20 01 28
> 
> Looking at the JS docs, they use params like below
> 
> Screenshot 2023-05-10 at 20 07 15
> 
> whereas the Kotlin docs seem to use the "Notes" field to describe parameters/options

As for the params, saw that but I didn't see it in the yml. Assumed it has to do with the `$ref: '@supabase/gotrue-js.GoTrueAdminApi.updateUserById'`?

**Comment by kevcodez** [comment]
> As for the params, saw that but I didn't see it in the yml. Assumed it has to do with the $ref: '@supabase/gotrue-js.GoTrueAdminApi.updateUserById'? (But wasn't quite sure what that means)

Maybe @gregnr can shed some light on this - again, definitely not a blocker and if we could adapt this in a future PR to further improve the docs, that's fine 👍

File: apps/docs/docs/ref/kotlin/installing.mdx (hunk 21 above)
Comment (gregnr) [reply apps/docs/docs/ref/kotlin/installing.mdx:5]: TBH, I'm not sure if this is actually used anywhere at the moment. Let's track in another issue: https://github.com/supabase/supabase/issues/14272

**Comment by gregnr** [comment]
> Looking at the JS docs, they use params like below
@kevcodez @jan-tennert Looks like these params are JS/TS only for now. Let's address this in a separate feature.

File: spec/supabase_kt_v0.yml (hunk 23 above)
Comment (gregnr) [reply spec/supabase_kt_v0.yml:95]: @jan-tennert Any ideas where this link came from originally? Also seeing it here:
https://github.com/supabase-community/supabase-kt/blob/c0b196dd4e12591d4a48a09ff9c564f02a5b9ac2/GoTrue/src/jvmMain/kotlin/io/github/jan/supabase/gotrue/GoTrueConfig.kt#L30

**Review by kevcodez — APPROVED** [review]
Found one more item with a missing description: "Column contains every element in a value"

Already going to approve though, looks great and thank you so much for the contribution, this is epic work!

**Review by gregnr — APPROVED** [review]
Amazing work @jan-tennert 🎉 🎉
